tests/utils/test_helpers.py

```diff
@@ -48,93 +50,95 @@ def _raise_fatal_failure():
         )
 
 
-def test_run_name_regex():
-    cases = [
-        {"value": "asd", "valid": True},
-        {"value": "Asd", "valid": True},
-        {"value": "AsA", "valid": True},
-        {"value": "As-123_2.8A", "valid": True},
-        {"value": "1As-123_2.8A5", "valid": True},
-        {
-            "value": "azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azs",
-            "valid": True,
-        },
-        {
+@pytest.mark.parametrize(
+    "value, valid",
+    [
+        ("asd", True),
+        ("Asd", True),
+        ("AsA", True),
+        ("As-123_2.8A", True),
+        ("1As-123_2.8A5", True),
+        (
+            "azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azs",
+            True,
+        ),
+        (
             # Invalid because the first letter is -
-            "value": "-As-123_2.8A",
-            "valid": False,
-        },
-        {
+            "-As-123_2.8A",
+            False,
+        ),
+        (
             # Invalid because the last letter is .
-            "value": "As-123_2.8A.",
-            "valid": False,
-        },
-        {
+            "As-123_2.8A.",
+            False,
+        ),
+        (
             # Invalid because $ is not allowed
-            "value": "As-123_2.8A$a",
-            "valid": False,
-        },
-        {
+            "As-123_2.8A$a",
+            False,
+        ),
+        (
             # Invalid because it's more then 63 characters
-            "value": "azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsx",
-            "valid": False,
-        },
-    ]
-    for case in cases:
-        try:
-            verify_field_regex("test_field", case["value"], mlrun.utils.regex.run_name)
-        except Exception:
-            if case["valid"]:
-                raise
-        else:
-            if not case["valid"]:
-                raise
+            "azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsxdcfvg-azsx",
+            False,
+        ),
+    ],
+)
+def test_run_name_regex(value, valid):
+    try:
+        verify_field_regex("test_field", value, mlrun.utils.regex.run_name)
+    except Exception:
+        if valid:
+            raise
+    else:
+        if not valid:
+            raise
```

Comment:
I think a much cleaner test here would be:
```suggestion
        assert verify_field_regex(
            "test_field",
            value,
            mlrun.utils.regex.run_name,
            raise_on_failure=False,
        ) == valid
```